core/src/core/block.rs

```diff
@@ -135,33 +135,21 @@ impl Default for BlockHeader {
 			output_root: ZERO_HASH,
 			range_proof_root: ZERO_HASH,
 			kernel_root: ZERO_HASH,
+			total_kernel_offset: BlindingFactor::zero(),
 			nonce: 0,
 			pow: Proof::zero(proof_size),
-			total_kernel_offset: BlindingFactor::zero(),
 		}
 	}
 }
 
 /// Serialization of a block header
 impl Writeable for BlockHeader {
 	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), ser::Error> {
-		ser_multiwrite!(
-			writer,
-			[write_u16, self.version],
-			[write_u64, self.height],
-			[write_fixed_bytes, &self.previous],
-			[write_i64, self.timestamp.to_timespec().sec],
-			[write_fixed_bytes, &self.output_root],
-			[write_fixed_bytes, &self.range_proof_root],
-			[write_fixed_bytes, &self.kernel_root]
-		);
-
-		try!(writer.write_u64(self.nonce));
-		try!(self.total_difficulty.write(writer));
-		try!(self.total_kernel_offset.write(writer));
-
-		if writer.serialization_mode() != ser::SerializationMode::Hash {
+		if writer.serialization_mode() == ser::SerializationMode::Hash {
 			try!(self.pow.write(writer));
```

Comment:
From eavesdropping on prior discussions it sounds the `pow` effectively encodes everything we care about - so we can just hash this (which is presumably faster)?

So effectively `block.hash() == header.hash() == pow.hash()`? Which has a nice feel to it.

Reply:
Yes ... all seems to slot nicely in place with this approach, and everything commits to exactly as much as it needs to and no more